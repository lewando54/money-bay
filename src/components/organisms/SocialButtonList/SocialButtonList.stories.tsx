/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react'

import SocialButtonList from './SocialButtonList'
import { create } from 'react-test-renderer'

const meta = {
    component: SocialButtonList,
    argTypes: {
        socials: { control: 'array' }
    }
} satisfies Meta<typeof SocialButtonList>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
    args: {
        socials: [
            { id: 0, color: '#3B5999', icon: 'facebook', href: 'https://google.pl/' },
            { id: 1, color: '#55ACEE', icon: 'twitter', href: 'https://google.pl/' },
            { id: 2, color: '#DD4B39', icon: 'google', href: 'https://google.pl/' }
        ],
        testId: 'sociallisttest'
    }
}