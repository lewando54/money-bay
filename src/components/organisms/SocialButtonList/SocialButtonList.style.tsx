import { StyleSheet } from 'react-native'

export default StyleSheet.create({
    div: {
        display: 'flex',
        width: '100%',
        justifyContent: 'center',
        gap: 10,
        flexDirection: 'row'
    }
})