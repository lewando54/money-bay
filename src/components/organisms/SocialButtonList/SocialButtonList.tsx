import React from 'react'
import { View } from 'react-native'

import { SocialButton } from '@atoms'
import { TiconProp } from 'src/components/atoms/SocialButton/SocialButton'

import socialButtonListStyle from './SocialButtonList.style'

export type TSocialsArray = Array<{ id: number, color: string, icon: TiconProp, href: URL['href'], testId?: string }>
interface ISocialButtonList {
  socials: TSocialsArray
  testId?: string
}

function SocialButtonList ({ socials, testId }: ISocialButtonList): React.ReactElement {
    const socialItems = socials.map(social =>
        <SocialButton size={28} key={social.id} color={social.color} href={social.href} icon={social.icon} testId={testId + social.id.toString()}/>
    )

    return (
        <View testID={testId} style={socialButtonListStyle.div}>
            {socialItems}
        </View>
    )
}

export default SocialButtonList
