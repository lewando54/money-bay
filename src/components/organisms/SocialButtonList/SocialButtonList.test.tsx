import React from 'react'
import { render } from '@testing-library/react-native'
import * as renderer from 'react-test-renderer'
import SocialButtonList, { TSocialsArray } from './SocialButtonList'

describe('SocialButtonList', () => {
    jest.useFakeTimers()
    const socials: TSocialsArray = [
        { id: 0, color: '#e90800', icon: 'google', href: 'https://google.pl/' },
        { id: 1, color: '#4768aa', icon: 'facebook', href: 'https://google.pl/' },
        { id: 2, color: '#0068c1', icon: 'linkedin', href: 'https://google.pl/' }
    ]

    it('should render correctly', () => {
        const tree = renderer.create(<SocialButtonList socials={socials} testId='test'/>)
        expect(tree).toMatchSnapshot()
    })

    it('should render icons in list', () => {
        const { getByTestId } = render(<SocialButtonList socials={socials} testId='test'/>)
        expect(getByTestId('test0')).toBeTruthy()
        expect(getByTestId('test1')).toBeTruthy()
        expect(getByTestId('test2')).toBeTruthy()
    })
})
