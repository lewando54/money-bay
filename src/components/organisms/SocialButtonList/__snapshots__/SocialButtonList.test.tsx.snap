// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialButtonList should render correctly 1`] = `
<View
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "gap": 10,
      "justifyContent": "center",
      "width": "100%",
    }
  }
  testID="test"
>
  <View
    animatedStyle={
      {
        "value": {
          "transform": [
            {
              "scale": 1,
            },
          ],
        },
      }
    }
    collapsable={false}
    style={
      {
        "transform": [
          {
            "scale": 1,
          },
        ],
      }
    }
    testID="test0"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "padding": 8,
          },
          {
            "backgroundColor": "#e90800",
            "borderRadius": 49,
            "height": 49,
            "width": 49,
          },
        ]
      }
      testID="pressable-social-button"
    >
      <Text />
    </View>
  </View>
  <View
    animatedStyle={
      {
        "value": {
          "transform": [
            {
              "scale": 1,
            },
          ],
        },
      }
    }
    collapsable={false}
    style={
      {
        "transform": [
          {
            "scale": 1,
          },
        ],
      }
    }
    testID="test1"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "padding": 8,
          },
          {
            "backgroundColor": "#4768aa",
            "borderRadius": 49,
            "height": 49,
            "width": 49,
          },
        ]
      }
      testID="pressable-social-button"
    >
      <Text />
    </View>
  </View>
  <View
    animatedStyle={
      {
        "value": {
          "transform": [
            {
              "scale": 1,
            },
          ],
        },
      }
    }
    collapsable={false}
    style={
      {
        "transform": [
          {
            "scale": 1,
          },
        ],
      }
    }
    testID="test2"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "padding": 8,
          },
          {
            "backgroundColor": "#0068c1",
            "borderRadius": 49,
            "height": 49,
            "width": 49,
          },
        ]
      }
      testID="pressable-social-button"
    >
      <Text />
    </View>
  </View>
</View>
`;
